platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/ResilientGradleBuildBuilder.java

```diff
@@ -0,0 +1,121 @@
+import org.gradle.plugins.ide.internal.tooling.model.DefaultGradleBuild;
+import org.gradle.tooling.internal.gradle.DefaultProjectIdentifier;
+import org.jspecify.annotations.NullMarked;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@NullMarked
+public class ResilientGradleBuildBuilder extends GradleBuildBuilder {
+    private final BuildIncludeListener failedIncludedBuildsRegistry;
+    private Map<BuildState, Failure> brokenBuilds = new HashMap<>();
+    private Map<SettingsInternal, Failure> brokenSettings = new HashMap<>();
+
+    public ResilientGradleBuildBuilder(
+        BuildStateRegistry buildStateRegistry,
+        BuildIncludeListener failedIncludedBuildsRegistry
+    ) {
+        super(buildStateRegistry);
+        this.failedIncludedBuildsRegistry = failedIncludedBuildsRegistry;
+    }
+
+    @Override
+    protected Throwable ensureProjectsLoaded(BuildState target) {
+        try {
+            target.ensureProjectsLoaded();
+            return null;
+        } catch (LocationAwareException e) {
+//            System.err.println(e.getMessage());
+            if (e.getCause() instanceof org.gradle.kotlin.dsl.support.ScriptCompilationException) {
+                brokenBuilds.putAll(failedIncludedBuildsRegistry.getBrokenBuilds());
+                brokenSettings.putAll(failedIncludedBuildsRegistry.getBrokenSettings());
+                return e.getCause();
+            }
+            throw e;
+        }
+    }
+
+    @Override
+    protected DefaultGradleBuild convert(BuildState targetBuild, Map<BuildState, DefaultGradleBuild> all) {
+        DefaultGradleBuild model = all.get(targetBuild);
+        if (model != null) {
+            return model;
+        }
+        model = new DefaultGradleBuild();
+        all.put(targetBuild, model);
+
+        // Make sure the project tree has been loaded and can be queried (but not necessarily configured)
+        ensureProjectsLoaded(targetBuild);
+
+        Failure failure = brokenBuilds.get(targetBuild);
+        if (failure != null) {
+            model.setFailure(failure.toString());
+        } else if (!brokenSettings.isEmpty()) {
+            Map.Entry<SettingsInternal, Failure> settingsEntry = brokenSettings.entrySet().iterator().next();
+            ProjectDescriptor rootProject = settingsEntry.getKey().getRootProject();
+            BasicGradleProject root = convertRoot(targetBuild, rootProject);
+            model.setRootProject(root);
+            model.addProject(root);
+            model.setFailure(settingsEntry.getValue().toString());
+        }
+
+        GradleInternal gradle = targetBuild.getMutableModel();
+        if (targetBuild.isProjectsLoaded()) {
+            addProjects(targetBuild, model);
+        }
+        try {
+            addFailedBuilds(targetBuild, all, model);
+            addIncludedBuilds(gradle, model, all);
+        } catch (IllegalStateException e) {
+            System.err.println("IllegalState " + e.getMessage());
```

Comment:
💅 I think we don't need to print errors, or if we do we should probably use Logger. But I guess user doesn't want to see these errors, since they are "expected" in resilient mode